toolkit/Toolkit/SqlManagedInstanceToolkit.cs

```diff
@@ -0,0 +1,196 @@
+                {
+                    addresses += addr + ";";
+
+                    if (cnt++ == 0)
+                    {
+                        IPAddress = addr.ToString();
+                    }
+                }
+
+                result.Add(string.Format("Hostname {0} was successfully resolved to: {1}.", hostname, addresses));
+            }
+            catch (SocketException se)
+            {
+                if (se.ErrorCode == 11001)
+                {
+                    result.Add(string.Format("Hostname {0} could not be found", hostname));
+                }
+                else
+                {
+                    result.Add(string.Format("DNS resolution of {0} thrown following error {1}.", hostname, se.ErrorCode));
+                }
+            }
+            catch (Exception ex)
+            {
+                result.Add(string.Format("DNS resolution of {0} thrown following exc {1}.", hostname, ex.Message));
+            }
+
+            // Check TCP ping
+            //
+            Ping ping = new Ping();
+            try
+            {
+                PingReply reply = ping.Send(hostname);
+                result.Add(string.Format("PING to {0} reported status {1}.", hostname, reply.Status));
+            }
+            catch (Exception ex)
+            {
+                result.Add(string.Format("PING to {0} thrown following exc {1}.", hostname, ex.Message));
+            }
+
+            // Check tcp connection to 443
+            //
+            if (IPAddress != null)
+            {
+                try
+                {
+                    Socket socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+
+                    socket.Connect(IPAddress, 443);
+
+                    if (socket.Connected)
+                    {
+                        result.Add(string.Format("TCP connect to {0}:443 was successful.", hostname));
+                    }
+                }
+                catch (SocketException se)
+                {
+                    if (se.ErrorCode == 10061)
+                    {
+                        result.Add(string.Format("TCP connect to {0}:443 failed because there were no listener. However the port is open.", hostname));
+                    }
+                    else
+                    {
+                        result.Add(string.Format("TCP connect to {0}:443 failed with {1}.", hostname, se.Message));
+                    }
+                }
+                catch (Exception ex)
+                {
+                    result.Add(string.Format("TCP connect to {0}:443 failed with {1}.", hostname, ex.Message));
+                }
+            }
+
+            if (executeWriteOperation.Value)
+            {
+                // Try writing to the file
+                //
+                string requestUri = string.Format("{0}/checktestblob_{1}?{2}", storageAccount.Value, DateTime.UtcNow.ToString("yyyyMMd_HHmmss"), sasToken);

```

Comment:
Long shot, but script executor might override the file if they have it named the same. It might be good to do list request before put, as that way we get additional validation that list is working properly and we can validate that the file we are about to create does not exist.

Reply:
let me add guid here and avoid listing the files :)